pyproject.toml

```diff
@@ -6,14 +6,15 @@ readme = "README.md"
 authors = [{name = "Sym Roe"}, {name = "Virginia Dooley"}]
 
 dependencies = [
-	"django>=3.2,<4.3",
-	"django-pipeline==3.0.0",
-	"markdown",
-	"django-localflavor==4.0",
-	"setuptools==69.2.0",
-	"whitenoise",
-	"pysass",
-	"jsmin<3.1",
+    "django>=3.2,<4.3",
+    "django-pipeline==3.0.0",
+    "markdown",
+    "django-localflavor==4.0",
+    "setuptools>=69.2.0",
+    "whitenoise",
+    "pysass",
+    "jsmin<3.1",
+    "pytidylib",
```

Comment:
pytidylib was removed as a dependency in https://github.com/DemocracyClub/dc_django_utils/pull/84 but I'm not sure this was intended. It is a package only used in https://github.com/DemocracyClub/dc_django_utils/blob/main/dc_utils/tests/helpers.py but given `validate_html_str()` is a test helper we want to expose to applications using this package (as opposed to being used to test the package itself), I think it is really a dependency of this package as the helper won't work without it.. unless we want to say client applications are responsible for pulling in `pytidylib` in their dev requirements if they want to use `validate_html_str()` because then it keeps it out of the prod runtime environment. Happy to back this commit out (and account for it in all the client apps when we upgrade) if so. I guess you could go either way on this..

Reply:
Given that we might use this in Lambda based applications, we don't want to be in a situation where we're forced to install `libtidy` on Lambda. I don't think we would be, as `pytidylib` will install without `libtidy`.

In other contexts, I don't think it matters to have `pytidylib` installed in a prod-like environment?

Reply:
Approved, but reckon having a chat about this would be useful?

Reply:
Gah I meant to talk about this point this morning and then forgot.

`pytidylib` is just bindings. Having `pytidylib` installed doesn't force us to install `libtidy`.

The only difference this makes is if you try to use the helpers in `dc_utils.tests.helpers`.

If we don't have `pytidylib` installed, then trying to import `dc_utils.tests.helpers` will throw `ModuleNotFoundError: No module named 'tidylib'`.

If you have `pytidylib` installed but not `libtidy`, then trying to call either of the functions in that module will raise `OSError: Could not load libtidy using any of these names: libtidy,libtidy.so,libtidy-0.99.so.0,cygtidy-0-99-0,tidylib,libtidy.dylib,tidy` but that's it.

So either way, if you want to use anything in that module you have to take another manual step to install `libtidy` so we could say its two and you also have to install `pytidylib`. But also, I don't think installing the bindings by default hurts anything. I'm kinda happy to go either way on it, but lets pick one..

Reply:
Can we:

1. Install the library by default
2. Catch the `libtidy` exception and ignore it (turn the functions into a no-op or log a warning

?

That way, if a system has installed `libtidy` then we're all good, otherwise there's a single step to install it, or limited issues to not having it?

Reply:
I don't think that's a good good idea for a test helper.

i.e: if we don't install libtidy (in CI or locally) and all my tests pass despite my html being broken, that seems like a bad thing we don't want

Reply:
Remember this is a file that the lambda functions aren't importing (if they were, they' be throwing `ModuleNotFoundError` as the package stands)

Reply:
I guess we could catch the `OSError` and re-raise it as an `AssertionError` but that kind of feels like it is just a layer of indirection.

Reply:
Another thing we could do is put in an optional group, so you can install `dc_django_utils[tidy]` if you want it and `dc_django_utils` if you don't.

Reply:
I'm still not sure about this, but given the above it seems reasonable to keep the Python dependency here, as it's not really doing much and doesn't add to the installed code all that much. We then allow it to raise whatever it raises when it can't find `libtidy` installed.

Does that make sense?

Reply:
Yep. That works for me. I'll merge this as is. If it causes a problem, we can revisit it.